### PR TITLE
[W02] Add last year's tests

### PR DESCRIPTION
**Pull request by antecrescent**
The [original repository](https://github.com/MaximilianAnzinger/gad2022-tests) is licensed under the MIT license.
The exercise remains unchanged and these tests still pass.

#### General

- [x] I locally pass **all** tests or calculated manually the correct solutions on paper.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://github.com/JohannesStoehr/gad23-tests/blob/main/CONTRIBUTING.md).
- [x] The newly added tests do not reveal the solutions dynamically calculating the solutions as expected in the exercise.

### Description

The exercise remains unchanged from last year and my solution still passes the Artemis tests.
Therefore I'd like to share last year's tests with you!

#### Review
- [x] Review 1
- [ ] Optional Review 2

### CODE REVIEW

File: w02/test/gad/binarysearch/BoundSearchTest.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+		return Stream.of(
+				Arguments.of(new int[]{0, 4, 4, 4, 4, 9, 10}, 2, true, 1),
+				Arguments.of(new int[]{0, 4, 4, 4, 4, 9, 10}, 11, true, -1),
+				Arguments.of(new int[]{0, 4, 4, 4, 4, 9, 10}, 11, false, 6),
+				Arguments.of(new int[]{1, 1, 4, 4, 4, 9, 10}, 1, false, 1)
+		);
+	}
+
+	@ParameterizedTest
+	@MethodSource({"simple", "randomValues", "sameValues", "valueNotInArray", "duplicatesInArray", "valuesOnArrayBoundaries"})
```
Comment (Flnn0) [inline w02/test/gad/binarysearch/BoundSearchTest.java:89]: I have run the tests locall and I pass all, but for better understanding which Argument is running could you please bringt it in the order of the MethodSource if it' possible for you.

Comment (antecrescent) [reply w02/test/gad/binarysearch/BoundSearchTest.java:89]: Fixed in [a4d01c2](https://github.com/JohannesStoehr/gad23-tests/pull/18/commits/a4d01c22db9f840e1fd8c35a42fbcb3b9710f5ab)

**Review by Flnn0 — APPROVED** [review]
I have run the tests locall and I pass all.

**Comment by antecrescent** [comment]
@JohannesStoehr This pull request is ready to be merged.

**Review by ITegs — APPROVED** [review]
lgtm
also passing locally

**Pull request merged**